jwst/badpix_selfcal/badpix_selfcal_step.py

```diff
@@ -0,0 +1,197 @@
+
+
+import warnings
+from ..stpipe import Step
+from . import badpix_selfcal
+import numpy as np
+from jwst import datamodels as dm
+
+import logging
+log = logging.getLogger(__name__)
+log.setLevel(logging.DEBUG)
+
+__all__ = ["BadpixSelfcalStep"]
+
+
+class BadpixSelfcalStep(Step):
+    """
+    BadpixSelfcalStep: Flags residual artifacts as bad pixels in the DQ array
+    of a JWST exposure using a median filter and percentile cutoffs.
+
+    All input exposures in the association file (or manually-provided bkg_list) are combined
+    into a single background model using a MIN operation. The bad pixels are then identified
+    using a median filter and percentile cutoffs, and applied to the science data by setting
+    the flagged pixels, errors, and variances to NaN
+    and the DQ flag to DO_NOT_USE + OTHER_BAD_PIXEL.
+    """
+
+    class_alias = "badpix_selfcal"
+    bkg_suffix = "badpix_selfcal"
+
+    spec = """
+    flagfrac_lower = float(default=0.001)  #fraction of pixels to flag on the low-flux end
```

Comment:
There are some range checks in `badpix_selfcal` that could be represented in this spec by setting `min` and `max`. Here's an example from resample:
https://github.com/spacetelescope/jwst/blob/master/jwst/resample/resample_step.py#L43
Would you update the spec to use whatever relevant constraints can be represented?